### ISSUE TITLE
Implement payload validation rules

### ISSUE THREAD

**Issue by codysoyland** [sigstore/protobuf-specs#381]
The protocol buffer files in this repo are annotated with [AIP-203 field behavior annotations](https://google.aip.dev/203), denoting fields that are marked as required. These annotations are not generally useful for payload validation, as described in [AIP-4223](https://google.aip.dev/client-libraries/4223):

> The reason for this is that the google.api.field_behavior annotation is primarily a machine-readable form of documentation, and not a configuration for payload validation.

In `sigstore-go`, we've discussed [adding additional validation](https://github.com/sigstore/sigstore-go/issues/63) to prevent nil-pointer panics, fail fast, and improve error messages. Ideally, these validation rules could be embedded in the protobuf files as opposed to implemented separately in each client language.

I came across a promising set of libraries designed for adding user-defined constraints to protobuf fields: [protovalidate](https://github.com/bufbuild/protovalidate/). It has highly-customizable rules and has Go, Java, and Python libraries.

I would like to propose that we add `protovalidate` annotations to the protobuf files in this repo, or if there are any alternative practices/libraries anybody would like to suggest.

**Comment by loosebazooka** [comment]
Yeah, I wouldn't mind removing sigstore-java's [custom validator](https://github.com/sigstore/sigstore-java/blob/main/sigstore-java/src/main/java/dev/sigstore/bundle/BundleVerifier.java).
